### PR TITLE
Add network flow #595

### PR DESCRIPTION
**Pull request by sonho00**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [X] **If creating a new file :**
  - [X] added links to it in the README files ?
  - [X] included tests with it ?
  - [X] added description (overview of algorithm, time and space complexity, and possible edge case) in docstrings ?

- [ ] **if done some changes :**
  - [ ] wrote short description in the PR explaining what the changes do ?
  - [ ] Fixes #[issue number] if related to any issue

- [ ] **other**

### CODE REVIEW

File: tests/test_graph.py [hunk 1]
```diff
@@ -74,4 +77,49 @@ def test_dijkstra(self):
...
+
+class TestMaximumFlow(unittest.TestCase):
+    """
+    Test for the file maximum_flow.py
+
+    Arguments:
+        unittest {[type]} -- [description]
+    """
+    def test_ford_fulkerson(self):
+        capacity = [[0]*7 for i in range(7)]
```
Comment (keon) [inline tests/test_graph.py]: please use easy to the format that is easy to recognize 
```
capacity = [
    [10, 10]
     ...
]
```

File: algorithms/graph/maximum_flow.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+    q.put(source)
+    level[source] = 0
+    while q.qsize():
+        front = q.get()
+        for i in range(v):
+            if level[i] == -1 and flow[front][i] < capacity[front][i]:
+                level[i] = level[front] + 1
+                q.put(i)
+    return level[sink] != -1
+def dinic_dfs(capacity, flow, level, idx, sink, work, f = 1<<63):
```
Comment (keon) [inline algorithms/graph/maximum_flow.py]: organize format a bit
(space between methods etc)

File: algorithms/graph/maximum_flow.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+        tmp = 0
+        q = queue.Queue()
+        visit = [False for i in range(v)]
+        par = [-1 for i in range(v)]
+        visit[source] = True
+        q.put((source, 1<<63))
+        while q.qsize():
+            front = q.get()
+            idx = front[0]
+            f = front[1]
```
Comment (keon) [inline algorithms/graph/maximum_flow.py]:
```suggestion
idx, f = front
```

Comment (keon) [reply algorithms/graph/maximum_flow.py]: also, comment on what do `f` and `p` represent

**Comment by sonho00** [comment]
Changed name of variable f and p to current_flow and parent respectively. Also, fixed minor format

**Pull request merged**